pkg/event_data_storage/accumulator.go

```diff
@@ -0,0 +1,253 @@
+package accumulator
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"sniffer/pkg/config"
+	"sniffer/pkg/ebpfeng"
+	evData "sniffer/pkg/ebpfev/v1"
+
+	logger "github.com/kubescape/go-logger"
+	"github.com/kubescape/go-logger/helpers"
+)
+
+const (
+	AccumulatorSize     = 10
+	DROP_EVENT_OCCURRED = "drop event occurred\n"
+)
+
+type containersEventStreamer struct {
+	streamDataChannelForContainerID map[string]chan evData.EventData
+	registerMutex                   sync.Mutex
+}
+
+type Accumulator struct {
+	data                            []map[string][]evData.EventData
+	syncReaderWriterData            sync.RWMutex
+	listOfFirstKeysInsertInEachSlot []string
+	cacheSize                       int
+	eventChannel                    chan *evData.EventData
+	containersData                  containersEventStreamer
+	ebpfEngine                      ebpfeng.EbpfEngineClient
+}
+
+type ContainerAccumulator struct {
+	dataChannel chan evData.EventData
+	containerID string
+}
+
+var nodeAgentContainerID string
+var accumulatorInstance *Accumulator
+var accumulatorInstanceLock = &sync.Mutex{}
+
+func newAccumulator() *Accumulator {
+	accumulatorInstance = &Accumulator{
+		cacheSize:                       AccumulatorSize,
+		data:                            make([]map[string][]evData.EventData, AccumulatorSize),
+		listOfFirstKeysInsertInEachSlot: make([]string, AccumulatorSize),
+		eventChannel:                    make(chan *evData.EventData),
+		containersData: containersEventStreamer{
+			streamDataChannelForContainerID: make(map[string]chan evData.EventData),
+		},
+	}
+
+	return accumulatorInstance
+}
+
+func GetAccumulator() *Accumulator {
+	if accumulatorInstance == nil {
+		accumulatorInstanceLock.Lock()
+		defer accumulatorInstanceLock.Unlock()
+		if accumulatorInstance == nil {
+			logger.L().Debug("Creating accumulatorInstance now.")
+			accumulatorInstance = newAccumulator()
+		}
+	}
+
+	return accumulatorInstance
+}
+
+func CreateContainerAccumulator(containerID string, dataChannel chan evData.EventData) *ContainerAccumulator {
+	return &ContainerAccumulator{
+		dataChannel: dataChannel,
+		containerID: containerID,
+	}
+}
+
+func (acc *Accumulator) createNewSlotInIndex(event *evData.EventData, index int) {
```

Comment:
```suggestion
func (acc *Accumulator) createNewSlot(event *evData.EventData, index int) {
```
But still, what does "create new slot" mean? You should call it: "add id", "insert id", etc. so it is clear what this function do from the application perspective and not how the function implements it.